test/util.test.ts

```diff
@@ -0,0 +1,120 @@
+'use strict';
+import {
+  getRanges,
+  getUserId,
+  handleError,
+  reduceProblem
+} from '../node-backend/util';
+
+describe('The utility', () => {
+  describe('getUser', () => {
+    const userId = 'user1';
+    const expectedResult = {id: userId};
+    it('should return the user if it is on the request', () => {
+      const request: any = {
+        user: {id: userId}
+      };
+      const result = getUserId(request);
+      expect(result).toEqual(expectedResult);
+    });
+
+    it('should return the user if it is on the session of the request', () => {
+      const userId = 'user1';
+      const request: any = {
+        session: {
+          user: {id: userId}
+        }
+      };
+      const result = getUserId(request);
+      expect(result).toEqual(expectedResult);
+    });
+
+    it('should throw an error if there is no user on the request', () => {
+      const request: any = {};
+      try {
+        getUserId(request);
+      } catch (error) {
+        expect(error).toBe('No user id found');
+      }
+    });
+  });
+
+  describe('reduceProblem', () => {
+    it('should reduce the problem to only the parts needed', () => {
+      const problem: any = {
+        prefs: 'some prefs',
+        criteria: {
+          critId1: {
+            id: 'critId1',
+            pvf: {},
+            scale: [1, 2],
+            title: 'crit 1 title'
+          }
+        }
+      };
+
+      const result = reduceProblem(problem);
+
+      const expectedResult = {
+        criteria: {
+          critId1: {
+            scale: [1, 2],
+            pvf: {},
+            title: 'crit 1 title'
+          }
+        },
+        prefs: problem.prefs
+      };
+      expect(expectedResult).toEqual(result);
+    });
+  });
+
+  describe('getRanges', () => {
+    it('should return the scales ranges from a problem', () => {
+      const problem: any = {
```

Comment:
no types in this test?

Reply:
Didn't want to write a full blown IProblem when the test only uses a part of it.